sparkmagic/sparkmagic/livyclientlib/linearretrypolicy.py

```diff
@@ -11,6 +11,8 @@ def __init__(self, seconds_to_sleep, max_retries):
         self._max_retries = max_retries
 
     def should_retry(self, status_code, error, retry_count):
+        if None in (status_code, retry_count):
```

Comment:
It was happening to me when connecting to an endpoint that didn't exist. `requests.get` would return `None` for `status_code` and stdout would return these strange opaque errors about `NoneType` in `linearretrypolicy.py`, which isn't very descriptive to the end user. I can fetch a screenshot of the situation if you'd like :)